secretstore/secretstore.go

```diff
@@ -0,0 +1,78 @@
+package secretstore
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type storeImpl interface {
+	Get(key string) (string, error)
+	Set(key, value string) error
+	List() ([]string, error)
+}
+
+type SecretStore struct {
+	Name     string `toml:"name"`
+	Service  string `toml:"service"`
+	Password string `toml:"password"`
+
+	store storeImpl
+}
+
+// Init initializes all internals of the secret-store
+func (s *SecretStore) Init() error {
+	// Remove the password from memory when leaving
+	defer func() {
+		s.Password = strings.Repeat("*", len(s.Password))
```

Comment:
I like this idea of removing the password from memory, but because strings are immutable in Go this most likely won't override the password but allocate a new string instead (correct me if I am wrong here). Doing some research, there is this package called [memguard](https://github.com/awnumar/memguard) that could be useful?